eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -499,4 +475,49 @@ object ChannelFlags {
   val Private: ChannelFlags = ChannelFlags(announceChannel = false)
   val Public: ChannelFlags = ChannelFlags(announceChannel = true)
 }
+
+// NB: channel data only matters (and should be stored in the DB) once we have committed funds.
+// If we disconnect or abort before that, we can completely forget the channel.
+sealed trait ChannelData {
```

Comment:
How about:

```scala
sealed trait ChannelData
sealed trait TransientChannelData extends ChannelData
sealed trait PersistentChannelData extends ChannelData
```

This  would allow you to define `case class WaitForFunding* extends TransientChannelData` and keep the corresponding fsm states lean?

Reply:
I can indeed introduce another trait for the transient state. That doesn't change the fact that I want to decouple this from the actual FSM state though, otherwise I won't be able to refactor the channel opening phase to a separate FSM.

When that happens, `Channel.scala` will contain a state `OPENING` that is transient but isn't in the same hierarchy as each state of the channel open protocol. On top of this, the channel opening phase does contain a mix of transient and persistent states (it becomes persistent as soon as we broadcast the funding tx).

Reply:
Okay, it's probably too early to make those design choices anyway.

Reply:
Adding this new trait to the existing PR does make sense though, it lets us group many of the `case class` together, I'll add it.